stdweb-derive/src/lib.rs

```diff
@@ -69,6 +70,18 @@ pub fn derive_reference_type( input: TokenStream ) -> TokenStream {
                         panic!( "The value of '#[reference(instance_of = ...)]' is not a string!" );
                     }
                 },
+                syn::NestedMeta::Meta( syn::Meta::NameValue( ref meta ) ) if meta.ident == "constraint" => {
+                    if let syn::Lit::Str( ref str ) = meta.lit {
+                        let val = str.value();
+                        let parts : Vec<&str> = val.splitn(2,"=").collect();
+                        if parts.len() != 2 {
+                            panic!("The value of '#[reference(..., constraint = ...)]' must be in the form of \"js_field=value\"" );
+                        }
+                        constraints.push(( parts[0].to_owned(), parts[1].to_owned()) );
```

Comment:
Could you also check that the field is a valid JS identifier? (It'd be fine to make sure that it starts with a letter and the rest contains only `[a-zA-Z0-9_]`.)

Reply:
The JS valid identifier check seems reasonable. As to the question of Option vs Vec, I think the prior comment about reference(constraint...) being separate from reference(instance_of...) is germane. If they are to be in a single line, then it should be a vec, else an Option is fine. ( I do mildly prefer the single line approach for reasons stated in my prior comment )